chat/serializers.py

```diff
@@ -0,0 +1,169 @@
+            raise serializers.ValidationError(
+                "Private room must have exactly 2 participants."
+            )
+        return data
+
+    def create(self, validated_data):
+        """
+        Create and persist a new Room instance.
+        """
+        room = Room(**validated_data)
+        room.save()
+        return room
+
+    def update(self, instance, validated_data):
+        """
+        Update an existing Room instance with validated data.
+        """
+        instance.name = validated_data.get('name', instance.name)
+        instance.is_group = validated_data.get('is_group', instance.is_group)
+        instance.participants = validated_data.get('participants', instance.participants)
+        instance.save()
+        return instance
+
+
+class MessageSerializer(serializers.Serializer):
+    """
+    Serializer for chat messages.
+
+    Fields:
+        - room (str): The name of the room where the message belongs.
+        - sender_id (str): ID of the user sending the message.
+        - receiver_id (str, optional): ID of the recipient (required for private messages).
+        - text (str): Message content (with length restrictions and forbidden words filtering).
+        - timestamp (datetime): Time of message creation (defaults to now in UTC).
+        - is_read (bool): Indicates if the message has been read.
+
+    Validation:
+        - Text must not be empty, contain forbidden words, or spam-like repeated characters.
+        - Sender must belong to the room.
+        - For private rooms, receiver_id is required and must be the other participant.
+        - For group rooms, receiver_id must be a participant if provided.
+    """
+
+    room = serializers.CharField()
+    sender_id = serializers.CharField()
+    receiver_id = serializers.CharField(required=False, allow_null=True)
+    text = serializers.CharField(
+        min_length=MIN_MESSAGE_LENGTH, max_length=MAX_MESSAGE_LENGTH
+    )
+    timestamp = serializers.DateTimeField(default=now)
+    is_read = serializers.BooleanField(default=False)
+
+    def validate_text(self, value):
+        """
+        Validate the content of the message.
+        """
+        if not value.strip():
+            raise serializers.ValidationError("Message text cannot be empty.")
+
+        lowered = value.lower()
+        forbidden_pattern = r'\b(?:' + '|'.join(
+            re.escape(word) for word in FORBIDDEN_WORDS_SET
+        ) + r')\b'
+        if re.search(forbidden_pattern, lowered):
+            raise serializers.ValidationError("Message contains forbidden content.")
+
+        if re.search(r"([^aeiou\s])\1{10,}", value, re.IGNORECASE):
+            raise serializers.ValidationError("Message looks like spam.")
+
+        return value
+
+    def validate(self, data):
+        """
+        Cross-field validation for message consistency with the room.
+        """
+        try:
+            room = Room.objects.get(name=data['room'])
```

Comment:
Direct access to 'room' without checking existence may cause KeyError. Use .get() and raise ValidationError if missing.
```
room_name = data.get('room')
if not room_name:
    raise serializers.ValidationError("Room name is required.")
```